network/src/protocols/discovery/mod.rs

```diff
@@ -42,18 +42,15 @@ const ANNOUNCE_INTERVAL: Duration = Duration::from_secs(3600 * 24);
 
 pub struct DiscoveryProtocol<M> {
     sessions: HashMap<SessionId, SessionState>,
-    dynamic_query_cycle: Option<Duration>,
+    announce_internal: Option<Duration>,
     addr_mgr: M,
-
-    check_interval: Option<Duration>,
 }
 
 impl<M: AddressManager> DiscoveryProtocol<M> {
     pub fn new(addr_mgr: M) -> DiscoveryProtocol<M> {
         DiscoveryProtocol {
             sessions: HashMap::default(),
-            dynamic_query_cycle: Some(Duration::from_secs(7)),
-            check_interval: None,
+            announce_internal: None,
```

Comment:
Maybe it can be changed to allow set `announce_internal` for testing

Reply:
Currently our test code does not have a scenario where this period needs to be set, it can be added later when it is used

Reply:
I found it's required in integration test, added a new config key, please review again, thanks